src/services/catalog.service.ts

```diff
@@ -157,6 +157,10 @@ export class CatalogService {
     return new TemplateItem(resource, this);
   }
 
+  public getPublisherSynonym(rawVendor: string): string {
+    return (this.constants.PUBLISHER_SYNONYMS ? (_.get(this.constants.PUBLISHER_SYNONYMS, rawVendor) as string) || rawVendor : rawVendor);
```

Comment:
You have to pass `rawVendor` to `_.get` in an array, otherwise special characters like `.` in `rawVendor` cause problems.

```ts
return (this.constants.PUBLISHER_SYNONYMS ? (_.get(this.constants.PUBLISHER_SYNONYMS, [rawVendor]) as string) || rawVendor : rawVendor);
```

But this is more concise and does the same thing

```ts
return _.get(this.constants, ['PUBLISHER_SYNONYMS', rawVendor]) as string || rawVendor;
```

Reply:
Thanks, fixed